File: p938.cpp
#include <bits/stdc++.h>
#include "TreeNode.hpp";

using namespace std;

class Solution {
public:
    void process(TreeNode* node, int& L, int& R, int& accumulate) {
        if(node == NULL)
            return;
        else if(node->val >= L && node->val <= R) {
            accumulate += node->val;
        }
        process(node->left, L, R, accumulate);
        process(node->right, L, R, accumulate);
    }
    int rangeSumBST(TreeNode* root, int L, int R) {
        int answer = 0;
        process(root, L, R, answer);
        return answer;
    }
};

File: p938.py
from DS.TreeNode import TreeNode

class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        if root is None:
            return 0
        ans = root.val if low <= root.val <= high else 0
        if root.val > low:
            ans += self.rangeSumBST(root.left, low, high)
        if root.val < high:
            ans += self.rangeSumBST(root.right, low, high)
        return ans
